bundle-ml/src/main/scala/ml/combust/bundle/BundleFile.scala

```diff
@@ -20,32 +20,43 @@ import scala.util.Try
   */
 object BundleFile {
   implicit def apply(uri: String): BundleFile = {
-    apply(new URI(uri))
+    apply(new URI(unbackslash(uri)))
   }
 
   implicit def apply(file: File): BundleFile = {
-    val uri = if(file.getPath.endsWith(".zip")) {
-      new URI(s"jar:file:${file.getAbsolutePath}")
+    val uri: String = if (file.getPath.endsWith(".zip")) {
+      s"jar:${file.toURI.toString}"
     } else {
-      new URI(s"file:$file")
+      file.toURI.toString
     }
 
     apply(uri)
   }
 
   implicit def apply(uri: URI): BundleFile = {
     val env = Map("create" -> "true").asJava
+    val uriSafe = new URI(unbackslash(uri.toString))
 
-    val (fs, path) = uri.getScheme match {
+    val (fs, path) = uriSafe.getScheme match {
       case "file" =>
-        (FileSystems.getDefault, FileSystems.getDefault.getPath(uri.getPath))
+        (FileSystems.getDefault, FileSystems.getDefault.getPath(uriSafe.getPath))
       case "jar" =>
-        val zfs = FileSystems.newFileSystem(uri, env)
+        val zfs = FileSystems.newFileSystem(uriSafe, env)
         (zfs, zfs.getPath("/"))
     }
 
     apply(fs, path)
   }
+
+  private def unbackslash(uri: String): String = {
+    // same logic as https://github.com/apache/commons-io/blob/master/src/main/java/org/apache/commons/io/FilenameUtils.java#L570
+    val BACKSLASH = '\\'
+    val NOT_FOUND = -1
```

Comment:
We shouldn't check for null URI, it should fail because this is definitely not expected input. Also, don't bother first checking for if backslash is in the string or not, just perform the replace.